qcengine/programs/nwchem/harvester.py

```diff
@@ -636,28 +634,36 @@ def harvest_outfile_pass(outtext):
                 re.MULTILINE,
             )
             if mobj:
-                homo = float(mobj.group(1)) * (10 ** (-1 * float(mobj.group(3))))
-                psivar["HOMO"] = np.array([round(homo, 10)])
-                mobj = re.search(
-                    r"Vector"
-                    + r"\s+"
-                    + r"%d" % (psivar["N ALPHA ELECTRONS"] + 1)
-                    + r"\s+"
-                    + r"Occ="
-                    + r".*"
-                    + r"\s+"
-                    + r"E="
-                    + r"([+-]?\s?\d+[.]\d+)"
-                    + r"[D]"
-                    + r"([+-])"
-                    + r"[0]"
-                    + r"(\d+)",
-                    outtext,
-                    re.MULTILINE,
-                )
+                if mobj.group(2) == "+":  # if "positve power eg."2.1225011D+01"
+                    homo = float(mobj.group(1)) * (10 ** (float(mobj.group(3))))
+                    psivar["HOMO"] = np.array([round(homo, 10)])
+                else:
+                    homo = float(mobj.group(1)) * (10 ** (-1 * float(mobj.group(3))))
+                    psivar["HOMO"] = np.array([round(homo, 10)])
```

Comment:
could you capture the sign, 0, and exponent together and `int` them, so that the if/else merges again?
```
>>> a="+01"
>>> 5**(int(a))
5
>>> a="-01"
>>> 5**(int(a))
0.2
```

Reply:
or grab the whole thing and replace `a.replace("D", "E")`
or I think NUMBER handles D/E. if you try it and it gives trouble, paste me the error

Reply:
Okay I made a change that captures the sign,0 and exponent together

Reply:
I captured the +-,0, and exponent and removed the if else blocks